src/Analyzer/Resolve/QueryAnalyzer.cpp

```diff
@@ -1141,6 +1144,43 @@ std::string QueryAnalyzer::rewriteAggregateFunctionNameIfNeeded(
     return result_aggregate_function_name;
 }
 
+/// A helper that builds and resolves `IF(isNull(lhs), lhs, has(rhs, lhs))`
+std::pair<QueryTreeNodePtr, ProjectionNames> QueryAnalyzer::makeNullSafeHas(
+    QueryTreeNodePtr array_arg,    // [1,2,number]
+    QueryTreeNodePtr element_arg,  // x (e.g. NULL)
+    const ProjectionNames & params_proj,
```

Comment:
I think `params_proj` is always empty, so it can be removed and replaced with `{}`